source/manual/howto-use-restored-db-in-app.html.md

```diff
@@ -0,0 +1,83 @@
+---
+owner_slack: "#govuk-2ndline-tech"
+title: How to use a restored database in an app
+section: Backups
+layout: manual_layout
+parent: "/manual.html"
+---
+
+Once you have [restored a database in AWS RDS](/manual/howto-backup-and-restore-in-aws-rds.html),
+you can now use this in your app.
+
+> The steps outlined below are for what to do during an incident where you may need to
+temporarily switch to use a backup. For a more permanent solution this will require
+work in Terraform as any new instances will still use the original database.
+
+### 1. Connect to the machine
+
+```
+gds govuk connect -e <environment> ssh <machine_name>
+```
+
+For example:
+
+```
+gds govuk connect -e integration ssh local_links_manager_db_admin
+```
+
+### 2. Get the root database credentials
+
+This is stored on the machine at /root/.pgpass which you can obtain by running:
+
+```
+sudo less /root/.pgpass
+```
+
+The credentials can also be found in govuk-secrets hieradata file e.g. https://github.com/alphagov/govuk-secrets/blob/main/puppet_aws/hieradata/blue/integration_credentials.yaml#L39
+
+Make note of the hostname, in this example it's `local-links-manager-postgres`. You will need this when connecting to the backup database in the next steps.
+
+### 3. Load a postgres console
+
+```
+sudo psql -U <user> -h <host> -d <database name>
+```
+
+In this example we want to connect to the database `local-links-manager_production`:
+
+```
+sudo psql -U aws_db_admin -h local-links-manager-postgres -d local-links-manager_production
+```
+
+### 4. Delete an old record
+
+This will help us determine whether we are using the backup database which will still
+contain this record.
+
+Select and delete an old user e.g.
+
+```
+select * from users where id=6;
+delete from users where id=6;
+DELETE 1
+```
```

Comment:
I think we can be less prescriptive here - sections 1-4 are all about making a change we can use to verify whether the DB switch has happened. I think that can be simplified to something like:

```md
## 1. Make a change to the database contents

Through the app's user interface, or via the app console or database console, make a change that you can use as a sense check to verify that the database switch has been successful.

For example, you might create a draft edition of something, or modify or delete a record. After switching to the restored database, your changes should be undone.
```

This has the advantage of being generic enough to apply to MySQL restores too - what you have so far will only work for Postgres.